### PR TITLE
Added don't show again button to obsolete system infobar

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/27122

W/o this, user will see this infobar whenever brave is launched.

Screenshot 2022-12-07 161035

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Launch Brave on Win7/8 and check obsolete system infobar with `don't show again` button is shown
2. Click `don't show again` button and check infobar is hidden
3. Relaunch and check this infobar is not shown anymore

### CODE REVIEW

**Comment by simonhong** [comment]
Set as draft because `kSuppressUnsupportedOSWarning` prefs should be cleared when user did os update.

**Comment by simonhong** [comment]
It's ready to review. PTAL :)

File: browser/brave_browser_process_impl.cc [hunk 1]
```diff
@@ -153,6 +155,14 @@ void BraveBrowserProcessImpl::Init() {
...
 
   InitSystemRequestHandlerCallback();
+
+#if !BUILDFLAG(IS_ANDROID)
+  if (!ObsoleteSystem::IsObsoleteNowOrSoon()) {
+    // Clear to show unsupported warning infobar again even if user
+    // suppressed it from previous os.
+    local_state()->ClearPref(prefs::kSuppressUnsupportedOSWarning);
+  }
+#endif
```
Comment (sangwoo108) [inline browser/brave_browser_process_impl.cc:165]: Is this code for the case where obsolete OS becomes supported later on?

**Review by sangwoo108 — APPROVED** [review]
LGTM. But I'm not quite sure if it's good to suppress this warning. I feel like users could be exposed to potential danger by using an obsolete version. So, I would prefer showing the warning even if it bothers them so that they could feel like updating the system. But it's just my persnoal preference.

File: browser/brave_browser_process_impl.cc (hunk 1 above)
Comment (simonhong) [reply browser/brave_browser_process_impl.cc:165]: When user dismissed this warning, that user could upgrade their os later.
And that os could be obsolete later. When it happens, this warning infobar should be visible again.
So, clear it here.

Comment (sangwoo108) [reply browser/brave_browser_process_impl.cc:165]: > that user could upgrade their os later.

Oh, great 👍 Thanks.

**Review by goodov — APPROVED** [review]

**Comment by kjozwiak** [comment]
> LGTM. But I'm not quite sure if it's good to suppress this warning. I feel like users could be exposed to potential danger by using an obsolete version - some security patches could be missing for these systems, I guess. So, I would prefer showing the warning even if it bothers them so that they could feel like updating the system. But it's just my persnoal preference.

Just adding some more context, users would also be able to see a warning regarding their Brave installs not getting updates via `brave://settings/help` as well whenever they go into `Settings` -> `About Brave`. Granted, not sure how many users would actually go there.

**Pull request merged**